packages/cli/src/commands/new2/new2.ts

```diff
@@ -0,0 +1,281 @@
+): Promise<TemplateLocation> {
+  const answer = await inquirer.prompt<{ name: TemplateLocation }>([
+    {
+      type: 'list',
+      name: 'name',
+      message: 'What do you want to create?',
+      choices: templates.map(template => {
+        return {
+          name: template.id,
+          value: template,
+        };
+      }),
+    },
+  ]);
+  return answer.name;
+}
+
+function parseTemplates(config: ConfigReader): TemplateLocation[] {
+  const templateLocations = config.getOptionalConfigArray('cli.templates');
+  let templates: TemplateLocation[] = [];
+
+  if (templateLocations) {
+    templateLocations.forEach(template => {
+      templates.push({
+        id: template.getString('id'),
+        type: template.getString('type'),
+        target: template.getString('target'),
+      });
+    });
+  }
+
+  if (config.getOptionalBoolean('cli.defaults')) {
+    templates = [...templates, ...defaultTemplates];
+  }
+
+  return templates;
+}
+
+async function addToPackages(targetDir: string) {
+  const pkgJson = await fs.readJson(
+    paths.resolveTargetRoot(`${targetDir}/package.json`),
+  );
+  const templateRole = pkgJson.backstage?.role;
+
+  const addToBackend = ['backend-plugin-module', 'backend-plugin'];
+  const addToFrontend = ['frontend-plugin'];
+
+  if (addToBackend.find(role => role === templateRole)) {
+    // 🚨
+  }
+  if (addToFrontend.find(role => role === templateRole)) {
+    // 🚨
+  }
+  return targetDir;
+}
+
+interface Template {
+  description?: string;
+  template: string;
+  type: string;
+  prompts?: ConfigurablePrompt[];
+}
+
+function verifyTemplate(contents: Template, location: string) {
+  if (!contents.template) {
+    throw new Error(
+      'Your CLI template yaml file should have a template path specified',
+    );
+  }
+  const templatePath = paths.resolveTargetRoot(
+    dirname(location),
+    contents.template,
+  );
+
+  if (!fs.existsSync(templatePath)) {
+    throw new Error(
+      `Your CLI template skeleton does not exist: ${templatePath}`,
+    );
+  }
+  if (!['plugin', 'package'].some(type => contents.type === type)) {
+    throw new Error(
+      'Your template must specify a type of either "plugin" or "package"',
+    );
+  }
+  return {
+    prompts: contents.prompts || [],
+    templatePath,
+    templateType: contents.type,
+  };
+}
+
+async function fetchTemplateInformation(template: TemplateLocation) {
+  const { type, target } = template;
+  if (type === 'file') {
+    const location = paths.resolveTargetRoot(target);
+
+    if (!fs.existsSync(location)) {
+      throw new Error(`Not Found: ${location}`);
+    }
+    const contents = parse(fs.readFileSync(location, 'utf-8'));
+    const { prompts, templatePath, templateType } = verifyTemplate(
+      contents,
+      paths.resolveTargetRoot(target),
+    );
+    return {
+      prompts,
+      templatePath,
+      templateType,
+    };
+  } else if (type === 'url') {
+    throw new Error('WIP'); // 🚨 TODO
```

Comment:
This is kinda tricky with the current design :p

Tbh I think we need to bite the bullet and list out all individual template files. I think it could be fine to have a simplified version for local templates, but if your template contents is defined as a single string it's treated as a directory and rejected when you try to use it via a URL.